### PR TITLE
Stubbing Metadata::Base HTTP requests with fixtures in order to handle TLS/SSL errors

### LINKED ISSUE #832: spec/features/metadata_panel_spec randomly fails
This does not need seem to fail consistently for me, but I've been finding that the following test failures can occur:

```ruby
1) Metadata tools when metadata references are available shows up as XML
     Failure/Error: expect(page).to have_css '.CodeRay'
       expected to find css ".CodeRay" within #<Capybara::Node::Element tag="div" path="/HTML/BODY[1]/DIV[3]/DIV[1]/DIV[1]/DIV[2]/DIV[1]"> but there were no matches
     # ./spec/features/metadata_panel_spec.rb:24:in `block (5 levels) in <top (required)>'
     # ./spec/features/metadata_panel_spec.rb:22:in `block (4 levels) in <top (required)>'
     # ./spec/features/metadata_panel_spec.rb:21:in `block (3 levels) in <top (required)>'
```

### PR DESCRIPTION
**Pull request by jrgriffiniii**
Resolves #832

### CODE REVIEW

**Comment by jrgriffiniii** [comment]
`cg357zz0321` is now failing also:
```bash
circleci@97f90fa3cbcd:~/geoblacklight$ curl "https://purl.stanford.edu/cg357zz0321.mods"
curl: (35) error:141A318A:SSL routines:tls_process_ske_dhe:dh key too small
```

**Comment by mejackreed** [comment]
Thanks for looking at this. I'll see if we have the options to upgrade our certs.

**Comment by jrgriffiniii** [comment]
I'm sorry, I'm just stubbing all of the responses.  If I can get these passing on https://github.com/geoblacklight/geoblacklight/compare/master...jrgriffiniii:issues-789-jrgriffiniii-circleci-parallel-test, that should clear https://github.com/geoblacklight/geoblacklight/pull/829 for review

**Comment by jrgriffiniii** [comment]
I'm going to please need this reviewed before I can proceed with https://github.com/geoblacklight/geoblacklight/pull/829

File: spec/lib/geoblacklight/metadata/base_spec.rb [hunk 1]
```diff
@@ -40,6 +40,21 @@
     end
   end
 
+  context 'when attempts to connect to an endpoint URL raise an OpenSSL error' do
+    subject { metadata.document }
+
+    before do
+      allow(Geoblacklight.logger).to receive(:error)
```
Comment (mejackreed) [inline spec/lib/geoblacklight/metadata/base_spec.rb]: Do we need this allow statement also here?

**Comment by jrgriffiniii** [comment]
Thank you for the merge, and sorry for https://github.com/geoblacklight/geoblacklight/pull/834#discussion_r310145045, it was an error from being more familiar with the `allow`/`have_received` syntax.

**Pull request merged**